rfcs/pagination-component.md

```diff
@@ -0,0 +1,55 @@
+# Pagination Component
+
+### Exports
+
+- `Pagination`
+- `StatefulPagination`
+- `StatefulContainer`
+- `StyledRoot`
+- `StyledBackButton`
+- `StyledNextButton`
+- `StyledMaxLabel`
+- `STATE_CHANGE_TYPE`
+
+### Pagination API
+
+- `numPages: number` - Required
+  Max number of pages
+- `currentPage: number` - Required
+  The current page
+- `prepositionLabel: string` - Optional
+  Defaults to "of"
+- `overrides: {PrevButton, NextButton, Select}` - Optional
+  Presentational override props, all are `React.ComponentType<*> | {component, props, style}`
+
+### StatefulContainer API
+
+- `numPages: number` - Required
+  Max number of pages
+- `initialState: {currentPage: number}` - Optional
```

Comment:
do we want to add props like

* onChange (page change callback)
* pageSize (items per page)?

I was looking at https://github.com/react-component/pagination#api

Reply:
I knew I missed something ;). Updated to add `onPageChange` handler as well as a couple of other ones.

I don't think we should handle complex situations like `pageSize` for now since the current design doesn't support it. My original thinking was that this component's responsibilities are:

* Show current and max number of pages
* Allow users to navigate 1 back/forward (buttons) or arbitrary page (dropdown)

How the application hooks into that behavior is up to the user. Thoughts?